app/models/status.rb

```diff
@@ -130,32 +130,43 @@ def not_in_filtered_languages(account)
       where.not(language: account.filtered_languages)
     end
 
-    def as_home_timeline(account)
+    def as_home_timeline(account, limit = nil, max_id = nil, since_id = nil)
```

Comment:
Stylistically, because there are now 4 (many) arguments, it could be useful to use kwargs, e.g. `account, limit: nil, max_id: nil, since_id: nil`, because then when calling the function you don't have to remember the exact order of the arguments, it's like a hash of options.